src/ca/noxid/lab/tile/MapPane.java

```diff
@@ -445,15 +446,20 @@ protected void drawTileTypes(Graphics2D g) {
 				int tileType;
 				int tile1 = dataHolder.getTile(j, i, 1);
 				int tile2 = dataHolder.getTile(j, i, 2);
+				int tileLayer = 1;
 				if (EditorApp.EDITOR_MODE == 0) {
 					if (tile1 != 0) {
 						tileType = dataHolder.calcPxa(tile1);
 					} else {
 						tileType = dataHolder.calcPxa(tile2);
+						tileLayer = 2;
 					}
 				} else {
 					tileType = dataHolder.calcPxa(tile2);
+					tileLayer = 2;
 				}
+				if (!visArray[tileLayer])
```

Comment:
not so sure on this change. for one, i'd imagine that one could want to not see the "physical" tiles of the tile layer while still showing the tile types and you can achieve the same effect of hiding both by unticking the "show tile types". 
secondly, if you're gonna skip rendering the whole layer it should be done outside the loop to save performance.